widget/entry_test.go

```diff
@@ -428,3 +428,34 @@ func TestPasswordEntry_Obfuscation(t *testing.T) {
 	assert.Equal(t, "Hié™שרה", entry.Text)
 	assert.Equal(t, "*******", entryRenderTexts(entry)[0].Text)
 }
+
+func TestEntry_Hide(t *testing.T) {
+	entry := NewEntry()
+	entry.SetText("test")
+
+	entry.Hide()
+	assert.Equal(t, entry.Visible(), false)
+
+	for _, child := range Renderer(entry).Objects() {
+		assert.Equal(t, child.Visible(), false)
+	}
+}
+
+func TestEntry_Show(t *testing.T) {
+	entry := NewEntry()
+
+	entry.Hide()
+
+	entry.Show()
+	assert.Equal(t, entry.Visible(), true)
+
+	for _, child := range Renderer(entry).Objects() {
+		assert.Equal(t, child.Visible(), true)
+	}
+
+	// Reperform the test with text and ensure placeholder is not visible
```

Comment:
It would be nice to do this another way, but I can't see how I can determine if a child is the placeholderProvider

Reply:
The main provider is `Renderer(entry).(*entryRenderer).text`.
The placeholder provider is `Renderer(entry).(*entryRenderer).text`.

Using WidgetRenderer.Objects() is usually not the best bet as (you have seen) it's just a list of graphical objects.